integration-test/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/KubernetesSuite.scala

```diff
@@ -66,20 +66,66 @@ private[spark] class KubernetesSuite extends FunSuite with BeforeAndAfterAll wit
     runSparkPiAndVerifyCompletion()
   }
 
+  test("Run SparkPi with a master URL without a scheme.") {
+    val url = kubernetesTestComponents.kubernetesClient.getMasterUrl
+    sparkAppConf.set("spark.master", s"k8s://${url.getHost}:${url.getPort}")
+    runSparkPiAndVerifyCompletion()
+  }
+
+  test("Run SparkPi with custom driver labels.") {
+    sparkAppConf
+      .set("spark.kubernetes.driver.label.foo", "foo-value")
+      .set("spark.kubernetes.driver.label.bar", "bar-value")
+    runSparkPiAndVerifyCompletion(driverPodChecker = (driverPod: Pod) => {
+      doBasicDriverPodCheck(driverPod)
+      assert(driverPod.getMetadata.getLabels.get("foo") === "foo-value")
+      assert(driverPod.getMetadata.getLabels.get("bar") === "bar-value")
+    })
+  }
+
+  test("Run SparkPi with custom driver annotations.") {
+    sparkAppConf
+      .set("spark.kubernetes.driver.annotation.foo", "foo-value")
+      .set("spark.kubernetes.driver.annotation.bar", "bar-value")
+    runSparkPiAndVerifyCompletion(driverPodChecker = (driverPod: Pod) => {
+      doBasicDriverPodCheck(driverPod)
+      assert(driverPod.getMetadata.getAnnotations.get("foo") === "foo-value")
+      assert(driverPod.getMetadata.getAnnotations.get("bar") === "bar-value")
+    })
+  }
+
+  test("Run SparkPi with custom driver environment variables.") {
```

Comment:
Can these be combined since you are merely modifying the sparkConf and checking the driverPod in both? Having too many tests will might take a really long time.

Reply:
Done.